forward_authentication_service/PreAuth/demo-preauth.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+query="$1"
+clientip="$(echo $query | awk -F '&' '{print $1;}' | awk -F '=' '{print $2;}')"
+gatewayname="$(echo $query | awk -F '&' '{print $2;}' | awk -F '=' '{print $2;}')"
```

Comment:
`awk -F '[&=]'` will save you an extra awk call here

Reply:
That does not work unfortunately.
A typical format of the query string is:
?clientip=192.168.1.68&gatewayname=My Gateway&redir=http://fred.com&username=fred&emailaddr=fred@fredco.com
Note the leading "?" and gatewayname can have whitespace.
I am open to more efficient suggestions :-D

Reply:
Passing an URL to a script for parsing is a dangerous idea in general. Most people will get it wrong and open the device of hijacking. Better use something like this: `./script.sh 'gatewayname' 'My Gateway'`. Also, the arguments have to be sanitised (in case they contain ' or \).